pageserver/src/tenant/timeline/walreceiver/walreceiver_connection.rs

```diff
@@ -140,6 +142,9 @@ pub(super) async fn handle_walreceiver_connection(
             }
             Ok(())
         }
+        // Enrich the log lines emitted by this closure with meaningful context.
+        // TODO: technically, this task outlives the surrounding function, so, the
+        // spans won't be properly nested.
```

Comment:
I'm 99.99% sure the cancellation is triggered or the connection is closed, but yeah, I guess technically.

Reply:
Yeah it is. But, just because cancellation is triggered doesn't mean we're _waiting_ for the task to exit. We're not doing that right now.

Reply:
On my list of shutdown stuff.